commit.go

```diff
@@ -36,6 +43,202 @@ type CommitOptions struct {
 	// ReportWriter is an io.Writer which will be used to log the writing
 	// of the new image.
 	ReportWriter io.Writer
+	// HistoryTimestamp is the timestamp used when creating new items in the
+	// image's history.  If unset, the current time will be used.
+	HistoryTimestamp *time.Time
+}
+
+// PushOptions can be used to alter how an image is copied somewhere.
+type PushOptions struct {
+	// Compression specifies the type of compression which is applied to
+	// layer blobs.  The default is to not use compression, but
+	// archive.Gzip is recommended.
+	Compression archive.Compression
+	// SignaturePolicyPath specifies an override location for the signature
+	// policy which should be used for verifying the new image as it is
+	// being written.  Except in specific circumstances, no value should be
+	// specified, indicating that the shared, system-wide default policy
+	// should be used.
+	SignaturePolicyPath string
+	// ReportWriter is an io.Writer which will be used to log the writing
+	// of the new image.
+	ReportWriter io.Writer
+	// Store is the local storage store which holds the source image.
+	Store storage.Store
+}
+
+// shallowCopy copies the most recent layer, the configuration, and the manifest from one image to another.
+// For local storage, which doesn't care about histories and the manifest's contents, that's sufficient, but
+// almost any other destination has higher expectations.
+func (b *Builder) shallowCopy(dest types.ImageReference, src types.ImageReference, systemContext *types.SystemContext) error {
```

Comment:
From my POV, \*shrug\* this is fragile but it can work just fine. Whether to open an issue still depends on how seriously you care about that kind of architectural cleanliness, _if_ you agree at all that centralizing the mechanisms for storing images in containers/storage would be cleaner.

I personally would lean towards not tracking this as an issue. If a real need to refactor arises, someone will know :)